library/src/data/index.js

```diff
@@ -41,20 +41,236 @@ window.CoNDeT.data = {
+
+      this.stateManager.setState(newState);
+    };
+
+    changeOrder = function (tableId) {};
+
+    addConnection = function (tableId, rowId, secondTableId) {
+      var newState = copyTables(this.state);
+      var editTable = newState[tableId];
+
+      for (var i = 0; i < editTable.rows.length; i++) {
+        if (editTable.rows[i].id == rowId) {
+          editTable.rows[i].connections.push(secondTableId);
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    removeConnection = function (tableId, rowId, secondTableId) {
+      var newState = copyTables(this.state);
+      var editTable = newState[tableId];
+
+      for (var i = 0; i < editTable.rows.length; i++) {
+        for (var j = 0; j < editTable.rows[i].connections.length; j++) {
+          if (
+            editTable.rows[i].id == rowId &&
+            editTable.rows[i].connections[j] == secondTableId
+          ) {
+            editTable.rows[i].connections.splice(j, 1);
+          }
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    replaceInArray = function (toReplace, newValues) {
+      if (newValues != []) {
+        for (var i = 0; i < newValues.length; i++) {
+          toReplace[i] = newValues[i];
+        }
+      }
+    };
+
+    removeColumnWithId = function (columns, rows, type, id) {
+      if (id != null && id <= columns.length) {
+        columns.splice(id, 1);
+        for (var i = 0; i < rows.length; i++) {
+          if (type == "conditions") {
+            for (var j = 0; j < rows[i].conditions.length; j++) {
+              if (rows[i].conditions[j][0] == id) {
+                rows[i].conditions.splice(j, 1);
+              }
+            }
+          } else if (type == "decisions") {
+            for (var j = 0; j < rows[i].decisions.length; j++) {
+              if (rows[i].decisions[j][0] == id) {
+                rows[i].decisions.splice(j, 1);
+              }
+            }
+          }
+        }
+      }
+    };
+
+    clone = function (toClone) {
+      if (typeof toClone === "object") {
+        var copyOfTables = {};
+        for (var attr in toClone) {
+          if (toClone.hasOwnProperty(attr)) {
+            if (Array.isArray(toClone[attr])) {
+              copyOfTables[attr] = [];
+              for (var j = 0; j < toClone[attr].length; j++) {
+                copyOfTables[attr][j] = clone(toClone[attr][j]);
+              }
+            } else if (typeof toClone[attr] === "object") {
+              copyOfTables[attr] = {};
+              for (var key in toClone[attr]) {
+                if (toClone[attr].hasOwnProperty(key)) {
+                  copyOfTables[attr][key] = clone(toClone[attr][key]);
+                }
+              }
+            } else {
+              copyOfTables[attr] = toClone[attr];
+            }
+          }
+        }
+      } else {
+        copyOfTables = toClone;
+      }
+
+      return copyOfTables;
+    };
+
+    copyTables = function (tables) {
```

Comment:
A czy to jest potrzebne? Nie można by uogólnić  `clone` po prostu? Ewentualnie nawet w clone można dać:
```
if (Array.isArray(toClone)) {
  var objectWithArray= clone( { array: toClone} );
  return objectWithArray.array;
}
```